python/src/util.rs

```diff
@@ -33,3 +37,15 @@ impl<T> BuilderWrapper<T> {
         Ok(result)
     }
 }
+
+pub fn parse_distance_type(distance_type: impl AsRef<str>) -> PyResult<DistanceType> {
+    match distance_type.as_ref() {
```

Comment:
do we want to make distance type lower case ?

Reply:
I did.  Distance type comparison is now case insensitive.